cypress-io/cypress
# fix: Remove unneeded Firefox headless hack (run ci)

- Closes https://github.com/cypress-io/cypress/issues/24900

### User facing changelog
Fixes an issue where recent versions of Firefox in headless mode would behave inconsistently with headed.

### Additional details
Original PR, closed by github when we deleted `release/12.0.0`

Back in 2021, Firefox had a quirk where in headless mode it would never gain focus; a workaround was added in https://github.com/cypress-io/cypress/commit/8a4848837aea88f79446c77eeac097b713a0d0a6, forcing the active tab into focus whenever an element in the document gained focus.

However, this results in inconsistent behavior between headed and headless mode in some circumstances. For example, a user might click on a button to open a dropdown, but then Cypress forces focus onto the tab as a whole, causing the dropdown to close again before it can be interacted with.

The workaround no longer appears to be necessary in recent versions of Cypress; both locally and in CI, removing it results in consistent and working behavior.

I tested this with Firefox 93, 97, 103 and 107, with `click.cy.ts`, `cypress/e2e/issues/1939_1940_2190.cy.js` (which was updated when this code was originally introduced), and a smattering of other driver tests. I couldn't tell you what changed in the intervening time, but it all appears to work. :woman_shrugging: 

I was unable to test versions before 93; Firefox 90 and 87 (which we claim to support) simply crash when run on my machine, even outside of Cypress. I can't get them to start at all. I did not try 92 or 91.

### Steps to test

See the newly added test for a minimal example.

Also see https://github.com/cypress-io/cypress/issues/24900 for an in-the-wild repro. Contact @BlueWinds if you want to run this example locally, she has the login information from the original issue reporter.

### How has the user experience changed?
No change; behavior is now consistent between headless and headed mode, going all the way back to Firefox 93.

### PR Tasks
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
